src/queries/function_parameter_count_changed.ron

```diff
@@ -0,0 +1,50 @@
+SemverQuery(
+    id: "function_parameter_count_changed",
+    human_readable_name: "pub fn parameter count changed",
+    description: "Parameter count of a function has changed.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#fn-change-arity"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output @tag
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        parameters @fold @transform(op: "count") @output @tag(name: "parameters")
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @filter(op: "=", value: ["%name"])
+
+                        parameters @fold @transform(op: "count") @filter(op: "!=", value: ["%parameters"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+    },
+    error_message: "Parameter count of a publicly-visible function has changed.",
+    per_result_error_template: Some("function {{join \"::\" path}}, previously in file {{span_filename}}:{{span_begin_line}}"),
```

Comment:
Whenever the same item exists in both the baseline and the current version, let's try to report span information for the current version whenever possible — that's a bit more convenient for the user of the tool since looking up the relevant code doesn't require checking out an older version of the code first.

Also, what are your thoughts on mentioning how the parameter count has changed in the per-result error message? Would that be useful for the user to see, or just noise?

Reply:
I think that in most cases it is just a bloat, as after receiving such an error a manual inspection of the parameters is required anyways. The only scenario where I think that could be helpful is when somebody is new to the programming and the rust language, and could maybe remove the self parameter without realising that it is indeed a parameter, but that is a pretty specific case :)

Reply:
Okay, now that you put it like that, I do like it. I would go with just printing the baseline vs current parameters count though.